packages/engine/tests/simulations/experiment.rs

```diff
@@ -0,0 +1,277 @@
+use std::{
+    collections::HashMap,
+    fs::{self, File},
+    io::BufReader,
+    path::{Path, PathBuf},
+};
+
+use error::{bail, ensure, report, Result, ResultExt};
+use hash_engine::utils::OutputFormat;
+use regex::Regex;
+use serde::{de::DeserializeOwned, Deserialize};
+use serde_json::Value;
+
+pub type AgentStates = Value;
+pub type Globals = Value;
+pub type Analysis = Value;
+
+#[derive(Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct ExpectedOutput {
+    #[serde(default)]
+    pub json_state: HashMap<String, AgentStates>,
+    #[serde(default)]
+    pub globals: Option<Globals>,
+    #[serde(default)]
+    pub analysis_outputs: Option<Analysis>,
+}
+
+pub enum Experiment {
+    Simple { experiment: String },
+    SingleRun { steps: u64 },
+}
+
+impl Experiment {
+    pub fn run(&self, project: impl AsRef<Path>) -> Result<Vec<(AgentStates, Globals, Analysis)>> {
+        let output = std::env::var("OUT_DIR").wrap_err("$OUT_DIR is not set")?;
+        let manifest_path =
+            std::env::var("CARGO_MANIFEST_DIR").wrap_err("$CARGO_MANIFEST_DIR is not set")?;
+        let manifest_path = Path::new(&manifest_path);
+
+        let mut cmd = std::process::Command::new(manifest_path.join("target/debug/cli"));
+        cmd.env("RUST_LOG", "trace")
```

Comment:
Well I'm suggesting we'd run everything as info because it is a **lot** faster (by orders of magnitude in my experience so far). Then if a test fails (not because of instability but because we've actually broken something), we could rerun it with trace to give better information to the user.

Reply:
Ah, now I got your point. Then I'll add a todo here.